### ISSUE TITLE
Can't create unital magma in a braided monoidal category

### ISSUE THREAD

**Issue by doctorn** [homotopy-io/homotopy-rs#713]
In the alpha, attempting to add a generator of the following type (where red is a scalar)
<img width="239" alt="Screenshot 2022-08-19 at 11 01 37" src="https://user-images.githubusercontent.com/6421560/185595628-4211a204-62c5-44b1-9bdf-7d039dcf4552.png">$\rightarrow$<img width="161" alt="Screenshot 2022-08-19 at 11 03 24" src="https://user-images.githubusercontent.com/6421560/185595681-2472e126-0267-4b7f-a4d3-5621c9cb5410.png">
results in the following panic:
```
panicked at 'Rewrite is malformed: [Cone(0, [Composition(Incompatible), Composition(Incompatible), Composition(Incompatible)])]', homotopy-core/src/rewrite.rs:595:42

Stack:

getImports/imports.wbg.__wbg_new_693216e109162396@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web.js:383:21
console_error_panic_hook::hook::h3442c3630cd546fa@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[1729]:0x237c42
core::ops::function::Fn::call::hb65e39c43681a605@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[4712]:0x2b5629
std::panicking::rust_panic_with_hook::h84feca33bd4bd229@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[2947]:0x29430d
std::panicking::begin_panic_handler::{{closure}}::hd2eacd3bb9ff1eab@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[3252]:0x2a1861
std::sys_common::backtrace::__rust_end_short_backtrace::h976699518d897fb1@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[4295]:0x2b4198
rust_begin_unwind@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[3844]:0x2b022c
core::panicking::panic_fmt::hc171d095bc4a492d@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[3849]:0x2b035b
core::result::unwrap_failed::h68ab818eb89182b6@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[3334]:0x2a4439
homotopy_core::rewrite::RewriteN::new::h1706e6a1e26a6bde@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[1331]:0x207245
homotopy_core::rewrite::Rewrite::cone_over_generator::h2dc4e770b2791e0e@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[364]:0xfc7d9
homotopy_core::diagram::DiagramN::from_generator::hd91b2133c2f9942c@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[587]:0x1633df
homotopy_web::model::proof::signature::Signature::create_generator::h3c54c6b01528232b@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[571]:0x15d58e
homotopy_web::model::proof::ProofState::update::hcba3119e5f01e4f5@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[206]:0x1f949
homotopy_web::model::State::update::h24a8596289ff2295@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[267]:0x9e842
<homotopy_web::app::App as yew::html::component::Component>::update::h5b27c766454aa6fe@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[427]:0x1201c8
<yew::html::component::lifecycle::UpdateRunner<COMP> as yew::scheduler::Runnable>::run::h79dcb9180f03adf7@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[619]:0x16e22f
std::thread::local::LocalKey<T>::with::h20f9d2963f4b335f@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[2098]:0x25b151
yew::scheduler::start::h5f2fec2fcc41133e@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[4711]:0x2b5621
yew::html::component::scope::Scope<COMP>::callback_with_passive::{{closure}}::h61955821611015f7@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[3309]:0x2a3777
yew::callback::Callback<IN>::emit::h9f9ffdccee68810f@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[3035]:0x2987a8
homotopy_web::app::sidebar::Sidebar::install_keyboard_shortcuts::{{closure}}::h69136a20847f9f12@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[1535]:0x221ccf
<dyn core::ops::function::FnMut<(A,)>+Output = R as wasm_bindgen::closure::WasmClosure>::describe::invoke::h56fa17c099c67e38@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web_bg.wasm:wasm-function[4100]:0x2b2dd2
__wbg_adapter_33@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web.js:258:10
real@https://homotopy-rs--homotopy-io-labelled-xvbfeqnk.web.app/homotopy_web.js:199:20
```

I don't know if this is related to #702.

**Comment by jamievicary** [comment]
Why is this talking about composition? We should not be doing any composition in alpha at all except in the wellformedness checker.

**Comment by calintat** [comment]
We only use composition in the wellformedness check, the error says the composition failed because the rewrites are incompatible.

**Comment by calintat** [comment]
Fixed by c889d98bbed200102c2cdaef9800fa93cbd159e9. Thanks for finding this Nathan!

**Comment by doctorn** [comment]
I'm still seeing the same panic :(

**Comment by calintat** [comment]
It works for me?

**Comment by calintat** [comment]
Oh wait red is a scalar?

**Comment by calintat** [comment]
Ok it's still broken. I reckon it's related to #705.

**Comment by doctorn** [comment]
(Sorry for the confusion)

**Comment by calintat** [comment]
Ironically this was also broken in 2D and I fixed that so I didn't even think to try it in 3D.

**Comment by calintat** [comment]
Fixed by #718.
